src/Core/Traits/StatefulData.php

```diff
@@ -207,9 +207,13 @@ public function setState(array $data)
 			// Compare DateTime Objects
 			} elseif ($value instanceof \DateTimeInterface && $this->$key instanceof \DateTimeInterface) {
 				$current_key = $this->$key;
-				$interval = $value->diff($current_key);
 
-				if ($interval->format('F') > 0) {
+                $stored_date_ts = date_timestamp_get($current_key);
```

Comment:
Super minor but I'd prefer we use `$current_key->getTimestamp()` since we already know we have an object.